yt_dlp/YoutubeDL.py

```diff
@@ -625,7 +625,7 @@ def __init__(self, params=None, auto_init=True):
                     'Overwriting params from "color" with "no_color"')
             self.params['color'] = 'no_color'
 
-        term_allow_color = os.environ.get('TERM', '').lower() != 'dumb'
+        term_allow_color = os.environ.get('TERM', '').lower() != 'dumb' and not os.getenv('NO_COLOR')
```

Comment:
I think this access should be `getenv`.
```suggestion
        term_allow_color = os.getenv('TERM', '').lower() != 'dumb' and not os.getenv('NO_COLOR')
```
But, thinking about it, `NO_COLOR` just implies using `'no_color'` as the policy, so that should result in separate variables (same as `yt-dlp --color no_color`)